smapply/api.py

```diff
@@ -0,0 +1,324 @@
+            response = getattr(self.session, method)(urljoin(BASE_API_URL, url_suffix), **kwargs)
+        response.raise_for_status()
+        return response
+
+
+def process_user(sma_user):
+    """
+    Create/update User and Profile model objects based on SMApply user info
+
+    Args:
+        sma_user (ReturnDict): Data from a fluidreview.serializers.UserSerializer object
+
+    Returns:
+        User: user modified or created by the function
+    """
+    user, _ = User.objects.get_or_create(
+        email__iexact=sma_user['email'],
+        defaults={'email': sma_user['email'], 'username': sma_user['email']}
+    )
+    profile, _ = Profile.objects.get_or_create(user=user)
+    if not profile.smapply_id:
+        profile.smapply_id = sma_user['id']
+        profile.name = sma_user['full_name']
+        profile.save()
+    return user
+
+
+def parse_webhook(webhook):
+    """
+    Attempt to load a WebhookRequestSMA body as JSON and assign its values to other attributes.
+
+    Args:
+        webhook (WebhookRequestSMA): WebhookRequestSMA instance
+
+    """
+    try:  # pylint:disable=too-many-nested-blocks
+        body_json = json.loads(smart_text(webhook.body))
+        field_mapping = {'id': 'submission_id', 'award': 'award_id', 'user_id': 'user_id'}
+        required_fields = field_mapping.keys()
+        if not set(required_fields).issubset(body_json.keys()):
+            raise SMApplyException("Missing required field(s)")
+        for att in required_fields:
+            if att in body_json:
+                if att in required_fields:
+                    if body_json[att]:
+                        setattr(webhook, field_mapping[att], int(body_json[att]))
+        parse_webhook_user(webhook)
+        webhook.status = WebhookParseStatus.SUCCEEDED
+    except:  # pylint: disable=bare-except
+        webhook.status = WebhookParseStatus.FAILED
+        log.exception('Webhook %s body is not valid JSON or has invalid/missing values.', webhook.id)
+    finally:
+        webhook.save()
+
+
+def parse_webhook_user(webhook):
+    """
+    Create/update User and Profile objects if necessary for a SMApply user id
+    Args:
+        webhook(WebhookRequestSMA): a WebhookRequestSMA object
+    """
+    if webhook.user_id is None:
+        raise SMApplyException('user_id is required in WebhookRequestSMA')
+    profile = Profile.objects.filter(smapply_id=webhook.user_id).first()
+    if not profile:
+        # Get user info from SMApply API (ensures that user id is real).
+        user_info = SMApplyAPI().get('/users/{}'.format(webhook.user_id)).json()
+        serializer = UserSerializer(data=user_info)
+        serializer.is_valid(raise_exception=True)
+        user = process_user(serializer.data)
+    else:
+        user = profile.user
+    if webhook.award_id is not None:
+        klass_info = SMApplyAPI().get('/awards/{}'.format(webhook.award_id)).json()
+        personal_price = klass_info.value
```

Comment:
Should this be `klass_info['price']`?

The webhook parsing succeeds if I change it to the above, otherwise I get an error:
```
web_1     | [2018-11-16 21:02:58] ERROR 232 [smapply.api] api.py:212 - [00ea0c37d5f8] - Webhook 4 body is not valid JSON or has invalid/missing values.
web_1     | Traceback (most recent call last):
web_1     |   File "./smapply/api.py", line 208, in parse_webhook
web_1     |     parse_webhook_user(webhook)
web_1     |   File "./smapply/api.py", line 237, in parse_webhook_user
web_1     |     personal_price = klass_info.value
web_1     | AttributeError: 'dict' object has no attribute 'value'
```

Reply:
haha, you are right! The docs say it is `value` 🤔